src/librustc_mir/transform/promote_consts.rs

```diff
@@ -266,8 +277,10 @@ impl<'a, 'tcx> Promoter<'a, 'tcx> {
         });
 
         // Inject the Rvalue or Call into the promoted MIR.
+        let dest = Lvalue::Temp(new_temp);
+        self.push(StatementKind::StorageLive(dest.clone()), source_info.span);
```

Comment:
It's an interesting problem. Constants can borrow things that outlive the evaluation of the constant, so in a sense, this would be valid. I had planned to emit `StorageDead` at the end of the promoted `Mir`, after the assignment to `return`, but then I remembered the borrow thing.
We could not have any `Storage{Live,Dead}` in constants, that would be fine with me, for now.